nix/nix.go

```diff
@@ -131,22 +133,32 @@ func (ctx *NixContext) GetMachines(deploymentPath string) (hosts []Host, err err
 	return hosts, nil
 }
 
-func (ctx *NixContext) BuildMachines(deploymentPath string, hosts []Host, nixArgs []string, nixBuildTargets string) (path string, err error) {
+func (ctx *NixContext) BuildMachines(deploymentPath string, hosts []Host, nixArgs []string, nixBuildTargets string) (resultPath string, err error) {
 	hostsArg := "["
 	for _, host := range hosts {
 		hostsArg += "\"" + host.TargetHost + "\" "
 	}
 	hostsArg += "]"
 
-	// create tmp dir for result link
-	tmpdir, err := ioutil.TempDir("", "morph-")
-	if err != nil {
-		return "", err
+	resultLinkPath :=
+	  fmt.Sprintf("%s/.gcroots/%s",
+                      path.Dir(deploymentPath),
+                      path.Base(deploymentPath))
+	if ctx.KeepGCRoot {
+	  if err = os.MkdirAll(path.Dir(resultLinkPath), 755) ; err != nil {
```

Comment:
@krav I assume you tested this?
Permissions need to be `0755` (leading zero), otherwise: `d-wxr----x 1 jth users    14 Mar 14 14:28 .gcroots`

Reply:
d1ed927